lib/src/components/NxTable/__tests__/NxTable.test.tsx

```diff
@@ -4,163 +4,507 @@
+      render(<NxTable><NxTable.Body emptyMessage="Empty message" isLoading /></NxTable>);
+      expect(screen.getByRole('rowgroup')).not.toHaveTextContent('Empty message');
+    });
+
+    it('does not show the emptyMessage when in error', function() {
+      render(<NxTable><NxTable.Body emptyMessage="Empty message" error="Errr" retryHandler={() => { }} /></NxTable>);
+
+      expect(screen.getByRole('rowgroup')).not.toHaveTextContent('Empty message');
+    });
+
+    it('removes the emptyMessage when children are added', async function() {
+      const { rerender } = render(<NxTable><NxTable.Body emptyMessage="Empty message"></NxTable.Body></NxTable>);
+
+      expect(screen.getByRole('rowgroup')).toHaveTextContent('Empty message');
+
+      rerender(
+        <NxTable>
+          <NxTable.Body emptyMessage="Empty message">
+            <NxTable.Row>
+              <NxTable.Cell />
+            </NxTable.Row>
+            <NxTable.Row>
+              <NxTable.Cell />
+            </NxTable.Row>
+          </NxTable.Body>
+        </NxTable>
+      );
 
-    expect(component.find('td')).toHaveProp('colSpan', 2);
-    expect(component.find('td')).toHaveClassName('nx-cell--meta-info');
+      await waitFor(() => expect(screen.getByRole('rowgroup')).not.toHaveTextContent('Empty message'));
+    });
   });
 
-  it('Puts the correct colSpan on an emptyMessage meta-info cell', async function() {
-    let component: ReactWrapper;
-    await act(async () => {
-      component = mount(
+  describe('NxTable.Row', function() {
+    it('renders cell instead of columnheader when isFilterHeader is set to true', function() {
+      render(
         <NxTable>
-          <NxTableHead>
-            <NxTableRow>
-              <NxTableCell>Foo</NxTableCell>
-              <NxTableCell>Bar</NxTableCell>
-            </NxTableRow>
-          </NxTableHead>
-          <NxTableBody emptyMessage="Empty" />
+          <NxTable.Head>
+            <NxTable.Row isFilterHeader={true}>
+              <NxTable.Cell>Foo</NxTable.Cell>
+            </NxTable.Row>
+          </NxTable.Head>
         </NxTable>
       );
+
+      expect(screen.getByRole('cell')).toBeInTheDocument();
+    });
+
+    it('sets the clickable row accessible name based on the text of its cells', function() {
+      render(
+        <NxTable>
+          <NxTable.Body>
+            <NxTable.Row isClickable={true}>
+              <NxTable.Cell>Foo</NxTable.Cell>
+              <NxTable.Cell>Bar</NxTable.Cell>
+            </NxTable.Row>
+          </NxTable.Body>
+        </NxTable>
+      );
+
+      expect(screen.getByRole('row')).toHaveAccessibleName('Foo Bar');
     });
-    component!.update();
 
-    expect(component!.find('td')).toHaveProp('colSpan', 2);
-    expect(component!.find('td')).toHaveClassName('nx-cell--meta-info');
-    expect(component!.find('td')).toHaveText('Empty');
+    it('sets the isFilterHeader flag into the RowContext, normalized to a boolean', function() {
```

Comment:
Another overt use of an impl-detail context to avoid

Reply:
> Another overt use of an impl-detail context to avoid

Removed it here: 2238321fbb01bf1a4b269b75bd00ccdc7d9156eb